File: Calculator/ios-native/CalculatorHub/Calculators/ROICalculator.swift
import Foundation

struct ROIResult {
    let returnAmount: Double
    let roiPercentage: Double
}

class ROICalculator {
    func calculate(initialInvestment: Double, finalValue: Double) -> ROIResult? {
        guard initialInvestment != 0 else { return nil }
        
        let gain = finalValue - initialInvestment
        let roi = (gain / initialInvestment) * 100.0

        return ROIResult(returnAmount: gain, roiPercentage: roi)
    }
}

File: Calculator/android-native/app/src/main/java/com/qaantum/calculatorhub/calculators/ROICalculator.kt
package com.qaantum.calculatorhub.calculators

data class ROIResult(
    val returnAmount: Double,
    val roiPercentage: Double
)

class ROICalculator {
    fun calculate(
        initialInvestment: Double,
        finalValue: Double
    ): ROIResult? {
        if (initialInvestment == 0.0) return null

        val gain = finalValue - initialInvestment
        val roi = (gain / initialInvestment) * 100.0

        return ROIResult(
            returnAmount = gain,
            roiPercentage = roi
        )
    }
}
